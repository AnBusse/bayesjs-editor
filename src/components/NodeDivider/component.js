import React from 'react';
import PropTypes from 'prop-types';

const NodeDivider = ({ y, width }) => (
  <path d={`M0,${y} h${width}`} stroke="#333" />
);

NodeDivider.propTypes = {
  y: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
};

export default NodeDivider;
