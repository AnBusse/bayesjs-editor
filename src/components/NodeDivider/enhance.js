import { pure } from 'recompose';

const enhance = pure;

export default enhance;
