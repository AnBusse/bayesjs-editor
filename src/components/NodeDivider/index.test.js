import React from 'react';
import { shallow } from 'enzyme';
import NodeDivider from './component';

const defaultProps = {
  y: 5,
  width: 150,
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<NodeDivider {...compProps} />);
};

describe('NodeDivider Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });
});
