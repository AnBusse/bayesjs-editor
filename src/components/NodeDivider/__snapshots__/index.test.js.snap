// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeDivider Component matches snapshot 1`] = `
<path
  d="M0,5 h150"
  stroke="#333"
/>
`;
